src/GLideNHQ/TxHiResCache.cpp

```diff
@@ -281,13 +283,39 @@ TxHiResCache::LoadResult TxHiResCache::loadHiResTextures(const wchar_t * dir_pat
 		pfname = strstr(fname, ident.c_str());
 		if (pfname != fname) pfname = 0;
 		if (pfname) {
-			if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X#%08X", &chksum, &fmt, &siz, &palchksum) == 4)
-				pfname += (ident.size() + CRCFMTSIZ_LEN + PALCRC_LEN);
-			else if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X", &chksum, &fmt, &siz) == 3)
-				pfname += (ident.size() + CRCFMTSIZ_LEN);
-			else
+			uint32_t length = 0;
+
+			if (sscanf(pfname + ident.size(), "#%08X#%01X#%01X#%08X", &chksum, &fmt, &siz, &palchksum) == 4) {
```

Comment:
pfname + ident.size() is used in several places. It's better to use a local variable, like:
```
const char * strName = pfname + ident.size();
if (sscanf(strName, "#%08X#%01X#%01X#%08X", &chksum, &fmt, &siz, &palchksum) == 4) {
```